src/mapeo-project.js

```diff
@@ -643,6 +711,16 @@ function extractEditableProjectSettings(projectDoc) {
   return result
 }
 
+// TODO: maybe a better signature than a bunch of any?
+/** @param {DataType<any,any,any,any,any>} dataType */
+async function deleteAll(dataType) {
+  const deletions = []
+  for (const { versionId } of await dataType.getMany()) {
+    deletions.push(dataType.delete(versionId))
+  }
+  return Promise.all(deletions)
+}
```

Comment:
Is there a way to do all of this in a single database transaction? It's possible that we'll get a list from `getMany` which will be different by the time we delete them.

Reply:
I don't think there's a way now of doing a bunch of deletions in a single transaction. I think this would imply changes to `src/datatype.js` and also probably the sqlite-indexer? Probably @gmaclennan can cheap in